### DISCUSSION TITLE
Integrating Select with Formik

### DISCUSSION THREAD

**Discussion by richbachman** [ariakit/ariakit#2104] [Questions] [answered]
I'm working on a new Select component for our design system, and would like to use the Ariakit Select as a base. I'm struggling with getting it to work with Formik, specifically with the `useFormik` hook. Are there any examples out there floating around? Thanks!

**Comment by richbachman** [comment]
I've made it this far, but am not able to get the new value into the `onSubmit` function. https://codesandbox.io/s/frosty-swartz-psj1is?file=/index.js

**Reply by richbachman** [reply]
I've updated the CodeSandbox with a working version. Unsure if thats the best way to handle it though. Any help will be greatly appreciated. Thanks!

**Reply by diegohaz** [reply]
That's correct. I'd just change `defaultValue` to `value`.

You can also get some inspiration from https://ariakit.org/examples/form-select

**Reply by richbachman** [reply]
@diegohaz great. Thanks for verifying.

**Comment by richbachman** [comment, marked as answer]
Final CodeSandbox version is here if anyone is interested: https://codesandbox.io/s/frosty-swartz-psj1is?file=/index.js
